input/tangy-timed.js

```diff
@@ -480,20 +485,40 @@ class TangyTimed extends PolymerElement {
 
     }
   }
-
+  shouldGridAutoStop() {
+    console.log(this.autoStop)
+    const isSetsEqual = (a, b) => a.size === b.size && [...a].every(value => b.has(value));
+    const tangyToggleButtons = [].slice.call(this.shadowRoot.querySelectorAll('tangy-toggle-button'))
+    if (!tangyToggleButtons[0].__data.pressed) {
+      return false;
+    } else {
+      const indexes = tangyToggleButtons.slice(0, this.autoStop).map((button, index) => index)
+      let pressedItemsIndex = [];
+      tangyToggleButtons.reduce((prev, curr, index) => {
+        if (curr.__data.pressed) {
+          pressedItemsIndex = [...pressedItemsIndex, index]
+        }
+      }, [])
+      return isSetsEqual(new Set(indexes), new Set(pressedItemsIndex))
+    }
+  }
   onTangyToggleButtonClick(event) {
 
     let tangyToggleButtons = [].slice.call(this.shadowRoot.querySelectorAll('tangy-toggle-button'))
     let inputElements = [].slice.call(this.querySelectorAll('[name]'))
+    if (this.shouldGridAutoStop()) {
```

Comment:
To prevent this function from running when it doesn't need to and make the code more clear, add `this.autoStop !== undefined && ...`.